bindings/python/tests/test_parse.py

```diff
@@ -163,3 +163,28 @@ def test_call_expr(tmp_path):
 
     assert var_decl["name"] == "anInt"
 
+
+def test_default_delete_constructor(tmp_path):
+    file_contents = """
+    class aClass {
+        aClass() = default;
+
+        // disable the copy constructor
+        aClass(double) = delete;
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    class_decl = parsed_info["members"][0]
+
+    default_constructor = class_decl["members"][0]
+
+    assert default_constructor["kind"] == "CONSTRUCTOR"
```

Comment:
The test is aimed at detecting when you should bind them. The checks should be geared towards that point:
* Detect when a default ctor needs to be added
* Detect when a ctor doesn't need to be added (aka has been deleted)

Reply:
This is code parsing, not binding generation. Seems to me he's testing twice that constructors (any implementation, `default` and `delete`d too) are bound, so I'd add a trivial-implementation one to make this complete or alternatively call this redundant.

Implicit constructors/copy operators seem (from a logical standpoint) something to be discovered after parsing, as what's not there might need to be there. [This](https://pybind11.readthedocs.io/en/stable/advanced/classes.html#brace-initialization) is the only mention I found of them in the pybind guide.

Reply:
My mistake for not referencing the comment: https://github.com/divmadan/pcl/pull/19#discussion_r463538295

Reply:
A point:
The default and delete ctors' representation in clang parsing is the same as the normal one, so we can decide whether we want them separately or not.

Reply:
Is it possible to specify in the test that the ctor is defaulted or deleted?

Reply:
> Is it possible to specify in the test that the ctor is defaulted or deleted?

As in? 
assert ctor type as defaulted/deleted?

Reply:
Yes. Right now, test code for both is the same, which it should not be. I have no issues in this conversation otherwise

Reply:
For default, yes. Via #32
Unfortunately, no test for deleted ctor available in clang.

Reply:
Put a note in the parsing code somewhere for this in that case. If @aponza is happy, go ahead and merge it

Reply:
Happy, had forgotten about the now linked comment/test